execute/executetest/source.go

```diff
@@ -133,9 +133,7 @@ func CreateAllocatingFromSource(spec plan.ProcedureSpec, id execute.DatasetID, a
 }
 
 func (s *AllocatingFromProcedureSpec) Run(ctx context.Context) {
-	if err := s.alloc.Allocate(s.ByteCount); err != nil {
-		panic(err)
-	}
+	_ = s.alloc.Allocate(s.ByteCount)
```

Comment:
Why are we ignoring the error now?

Can you add a comment explaining why its safe to do so?

Reply:
It's mostly that `Allocate` doesn't return an error anymore. That got changed in another PR so this was just incorrect and I found it. I changed the interface to match arrow's so this returns a `[]byte` and it panics if there was an error.

Reply:
Ok, why do we not need the returned `[]byte`?

Reply:
There's nowhere for it to be returned to. This is just a special source for testing that the query panics and recovers when you overallocate.

Reply:
ok, that makes sense, can you add that as a comment?

Reply:
It's documented here:

```
// AllocatingFromProcedureSpec is a procedure spec AND an execution node
// that allocates ByteCount bytes during execution.
type AllocatingFromProcedureSpec struct {
    ByteCount int

    alloc *memory.Allocator
    ts    []execute.Transformation
}
```

Reply:
That comment doesn't really explain this 

>There's nowhere for it to be returned to. This is just a special source for testing that the query panics and recovers when you overallocate.

I am probably missing some context that makes this obvious, but since I am missing it its likely others are too, as such I'd like to see a comment added explaining why it is safe to ignore the returned value.

Reply:
Updated.